app/backend/src/couchers/servicers/auth.py

```diff
@@ -94,6 +94,9 @@ def _create_session(self, context, session, user, long_lived):
         if user.is_banned:
             context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.ACCOUNT_SUSPENDED)
 
+        if user.is_deleted:
+            context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.ACCOUNT_DELETED)
```

Comment:
This should behave as if the user didn't exist (so this needs to be handled before `_create_session`, I think). Just do `is_visible` in the search for the user in login?

Reply:
As it stands now, that would catch banned users as well.

Login uses "get_user_by_field". We could add it there? Or else change login to a query?